### ISSUE TITLE
Performance regression with 2023.2.11 Pre-Release

### ISSUE THREAD

**Issue by Gobot1234** [microsoft/pylance-release#3901]
**Describe the bug**
Performance when type checking is really slow

**To Reproduce**
I was editing some code in https://github.com/Gobot1234/steam.py/blob/main/steam/state.py and noticed that the LSP isn't updating as fast as it normally does leading to situations like
![image](https://user-images.githubusercontent.com/50501825/216403408-3a894694-ee22-4da5-a3df-100b9a216040.png) 
happening for multiple seconds and I go check my activity monitor and see that code is pinned at 100% 
![image](https://user-images.githubusercontent.com/50501825/216404375-9985a023-8e13-46bd-ae1b-77ab2ec6a68e.png)

**Expected behavior**
Not having to restart the LSP to get type hint info.

The full code I added was
```py
    async def handle_chat_update_from_message(
        self,
        chat_group: ChatGroup,
    ) -> None:
        ...

    async def handle_chat_message(self, msg: chat.IncomingChatMessageNotification) -> None:
        try:
            destination = self._chat_groups[ChatGroupID(msg.chat_group_id)]
        except KeyError:
            return log.debug(f"Got a message for a chat we aren't in {msg.chat_group_id}")

        channel = destination._channels[ChatID(msg.chat_id)]
        channel._update(msg)

        if msg.server_message:
            await self.handle_chat_update_from_message(chat_group)
            return

        (message_cls,) = channel._type_args
        message = message_cls(
            proto=msg,
            channel=channel,  # type: ignore  # type checkers aren't able to figure out this is ok
            author=destination._maybe_member(ID64(msg.steamid_sender)),
        )
        channel.last_message = message  # type: ignore  # same as above
        self._messages.append(message)
        self.dispatch("message", message)
```

Is there anything else you need to debug the issue further?

**VS Code extension or command-line**
pylance v2023.2.11 Pre-Release

**Comment by rchiodo** [comment]
Thanks. I could definitely short circuit that code path when there are no fixtures in a file, but it should be caching that logic. So it should happen once and just need to update itself for the file being typed into.

To repro, can I just open the https://github.com/Gobot1234/steam.py/blob/main/steam/state.py file and start editing? It looks like you also inlay hints active.

**Comment by Gobot1234** [comment]
> @Gobot1234, I just noticed that you're making use of TypeVar defaults (PEP 696) in this code base. Nice!

I'd hope I was one of the first to, considering I needed things to test my mypy fork on :p

> To repro, can I just open the [Gobot1234/steam.py@main/steam/state.py](https://github.com/Gobot1234/steam.py/blob/main/steam/state.py?rgh-link-date=2023-02-02T21%3A40%3A39Z) file and start editing? It looks like you also have inlay hints active.

I don't think the inlays will make any difference here, so yes

**Comment by rchiodo** [comment]
I can repro for sure. Completions take like 7 seconds to come up in that file.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.2.21, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023212-2-February-2023-PreRelease)

**Comment by debonte** [comment]
@arogozhnikov, please file a new issue.
